File: src/F#/Algorithms/InsertionSort.fs
namespace Algorithms

type InsertionSort =
    static member Sort (arr : 'a[]) =
        for i = 1 to arr.Length - 1 do
            let key = arr.[i]
            let mutable j = i - 1
            while j >= 0 && arr.[j] > key do
                arr.[j+1] <- arr.[j] 
                j <- j-1
            arr.[j+1] <- key

File: src/TS/src/Algorithms/InsertionSort/InsertionSort.ts
export function InsertionSort<T>(arr: T[]) {
    for (let i = 1; i < arr.length; i++) {
        let key = arr[i];
        let index = i - 1;
        while (index >= 0 && arr[index] > key) {
            arr[index + 1] = arr[index];
            index--;
        }
        arr[index + 1] = key;
    }
}